sunpy/io/file_tools.py

```diff
@@ -214,14 +214,16 @@ def detect_filetype(filepath):
         cdf_magic_number = fp.read(4).hex()
 
     # FITS
-    # Check the extensions to see if it is a gzipped FITS file
-    filepath_rest_ext1, ext1 = os.path.splitext(filepath)
-    _, ext2 = os.path.splitext(filepath_rest_ext1)
-
-    gzip_extensions = [".gz"]
-    fits_extensions = [".fts", ".fit", ".fits"]
-    if (ext1 in gzip_extensions and ext2 in fits_extensions):
-        return 'fits'
+    # Checks for gzip signature. 
+    # If found, decompresses first few bytes and checks for FITS
+    if cdf_magic_number[:4] == '1f8b':
```

Comment:
I would rather avoid using the `cdf_magic_number` variable here as it's semantically intended for detecting CDF files. In the code above `first80` is intended for FITS. Astropy detect with an extra byte [in this code](https://github.com/astropy/astropy/blob/f1620baa9f0a5228cac0c8bed22f42e4f4243896/astropy/io/fits/file.py#LL98C14-L98C15), so maybe we should check that too?
```suggestion
    if first80[:3] == b"\x1f\x8b\x08":
```
They also support PKZIP and BZIP2, which I don't think sunpy will detect, but maybe they are not as relevant to solar data though?

Reply:
I've not seen solar data with those but doesn't mean we won't in the future.